Intersect.Server/Networking/PacketHandler.cs

```diff
@@ -787,28 +788,23 @@ public void HandlePacket(Client client, ChatMsgPacket packet)
                     return;
                 }
 
+                Color chatColor = CustomColors.Chat.LocalChat;
                 if (client?.Power.IsAdmin ?? false)
                 {
-                    PacketSender.SendProximityMsg(
-                        Strings.Chat.local.ToString(player.Name, msg), ChatMessageType.Local, player.MapId, CustomColors.Chat.AdminLocalChat,
-                        player.Name
-                    );
+                    chatColor = CustomColors.Chat.AdminLocalChat;
                 }
                 else if (client?.Power.IsModerator ?? false)
                 {
-                    PacketSender.SendProximityMsg(
-                        Strings.Chat.local.ToString(player.Name, msg), ChatMessageType.Local, player.MapId, CustomColors.Chat.ModLocalChat,
-                        player.Name
-                    );
-                }
-                else
-                {
-                    PacketSender.SendProximityMsg(
-                        Strings.Chat.local.ToString(player.Name, msg), ChatMessageType.Local, player.MapId, CustomColors.Chat.LocalChat,
-                        player.Name
-                    );
+                    chatColor = CustomColors.Chat.ModLocalChat;
                 }
 
+                PacketSender.LogChatMsg(player, msg, ChatMessageType.Local);
+                PacketSender.SendProximityMsg(
+                    Strings.Chat.local.ToString(player.Name, msg), ChatMessageType.Local, player.MapId, chatColor,
+                    player.Name
+                );
+
+
                 PacketSender.SendChatBubble(player.Id, (int) EntityTypes.GlobalEntity, msg, player.MapId);
```

Comment:
I did this pattern a few times to remove duplicated code. No one to have a full chat message each time if all you are changing is the color. Also you can set a default case instead of the else to save code.